packages/react/src/components/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,174 @@
+          | React.KeyboardEvent<HTMLButtonElement>
+      ) => {
+        // istanbul ignore else
+        if (!event.defaultPrevented) {
+          setOpen(!open);
+          setFocusStrategy('first');
+        }
+      },
+      [open]
+    );
+
+    const handleTriggerKeyDown = useCallback(
+      (event: React.KeyboardEvent<HTMLButtonElement>) => {
+        // istanbul ignore else
+        if ([ArrowDown, ArrowUp].includes(event.key) && !open) {
+          // prevent page from scrolling if the user triggers the action menu
+          // via an "ArrowDown" key press
+          event.preventDefault();
+          // allow other functions that may consume the event after
+          // default is prevented to perform as normal
+          event.defaultPrevented = false;
+          setFocusStrategy(event.key === ArrowUp ? 'last' : 'first');
+          setOpen(true);
+        }
+      },
+      [open]
+    );
+
+    const triggerProps: React.HTMLAttributes<HTMLButtonElement> =
+      useMemo(() => {
+        return {
+          ref: triggerRef,
+          id: triggerId,
+          onClick: handleTriggerClick,
+          onKeyDown: handleTriggerKeyDown,
+          'aria-expanded': open,
+          'aria-haspopup': 'menu'
+        };
+      }, [handleTriggerClick, open]);
+
+    const actionMenuTrigger = React.isValidElement(trigger)
+      ? React.cloneElement(trigger, triggerProps)
+      : (trigger as ActionMenuTriggerFunction)(triggerProps, open);
+
+    const handleClickOutside = useCallback((event: MouseEvent | TouchEvent) => {
+      if (
+        !actionMenuRef.current?.contains(event.target as HTMLElement) &&
+        !triggerRef.current?.contains(event.target as HTMLElement)
+      ) {
+        setOpen(false);
+      }
+    }, []);
+
+    const handleAction = useCallback(
+      (key: string, event: onActionEvent) => {
+        // istanbul ignore else
+        if (!event.defaultPrevented) {
+          setOpen(false);
+        }
+
+        const { onAction } = actionMenuList.props;
+        if (typeof onAction === 'function') {
+          onAction(key, event);
+        }
+      },
+      [actionMenuList.props.onAction]
+    );
+
+    useEffect(() => {
+      if (open) {
+        actionMenuListRef.current?.focus();
+      } else if (actionMenuListRef.current?.contains(document.activeElement)) {
+        triggerRef.current?.focus();
+      }
+    }, [open]);
+
+    return (
+      <>
+        <ClickOutsideListener
+          onClickOutside={handleClickOutside}
+          mouseEvent={open ? undefined : false}
+          touchEvent={open ? undefined : false}
+        >
+          {actionMenuTrigger}
+        </ClickOutsideListener>
+        <AnchoredOverlay
+          ref={actionMenuRef}
+          role="presentation"
+          className={classnames('ActionMenu', className)}
+          open={open}
+          onOpenChange={setOpen}
+          target={triggerRef}
+          placement={placement}
+          offset={4}
```

Comment:
I think we want to set this to 8px.

This PR:
<img width="266" alt="Screenshot 2025-05-08 at 9 00 05 PM" src="https://github.com/user-attachments/assets/1818cf97-d974-411b-a511-2e60f6200bf0" />

Figma:
https://www.figma.com/design/CEFVdiecqDjLSjhorjHUzI/Product-Foundations--Cauldron--Library?node-id=4840-1786&p=f&m=dev
<img width="311" alt="Screenshot 2025-05-08 at 9 01 35 PM" src="https://github.com/user-attachments/assets/234bce07-b724-4ad5-af60-f21ca71a7f1c" />

Reply:
I need to check with @ezirlingerDQ on this, I think figma is offset wrong because this is assuming that the hover state will be present and 8px is a very large gap between the trigger and the action menu.

Reply:
Confirmed this should be 4px. Eric will update soon.